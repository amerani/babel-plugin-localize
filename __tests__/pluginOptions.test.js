const babel = require('@babel/core');
const plugin = require('../src/index');

const example = `
export default () => (
    <>
        <p>hello world</p>
        <Title id="titleId" label="lorem ipsum" />
    </>
);
`;

test('should set key prefix', () => {
  const { code } = babel.transform(example, {
    plugins: [[plugin, {
      keyPrefix: 'test',
    }]],
  });
  expect(code).toMatchSnapshot();
});

test('should set localizer', () => {
  const { code } = babel.transform(example, {
    plugins: [[plugin, {
      localizer: 'testLocalizer',
    }]],
  });
  expect(code).toMatchSnapshot();
});

test('should set localizerSource', () => {
  const { code } = babel.transform(example, {
    plugins: [[plugin, {
      localizerSource: 'testlocalizerSource',
    }]],
  });
  expect(code).toMatchSnapshot();
});

test('should set localizerBinding', () => {
  const { code } = babel.transform(example, {
    plugins: [[plugin, {
      localizerBinding: 'testlocalizerBinding',
    }]],
  });
  expect(code).toMatchSnapshot();
});

test('should set keyMapIdentifier', () => {
  const { code } = babel.transform(example, {
    plugins: [[plugin, {
      keyMapIdentifier: 'testKeyMapIdentifier',
    }]],
  });
  expect(code).toMatchSnapshot();
});
