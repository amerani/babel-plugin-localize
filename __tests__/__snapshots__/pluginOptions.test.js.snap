// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should set key prefix 1`] = `
"import { localize } from \\"./localizer\\";
export default (() => <>
        <p>{localize(\\"test0\\")}</p>
        <Title id=\\"titleId\\" label=\\"lorem ipsum\\" />
    </>);
export const localizeKeyMap = {
  test0: \\"hello world\\"
};"
`;

exports[`should set keyMapIdentifier 1`] = `
"import { localize } from \\"./localizer\\";
export default (() => <>
        <p>{localize(\\"4\\")}</p>
        <Title id=\\"titleId\\" label=\\"lorem ipsum\\" />
    </>);
export const testKeyMapIdentifier = {
  4: \\"hello world\\"
};"
`;

exports[`should set localizer 1`] = `
"import { testLocalizer } from \\"./localizer\\";
export default (() => <>
        <p>{testLocalizer(\\"1\\")}</p>
        <Title id=\\"titleId\\" label=\\"lorem ipsum\\" />
    </>);
export const localizeKeyMap = {
  1: \\"hello world\\"
};"
`;

exports[`should set localizerBinding 1`] = `
"import { localize } from \\"./localizer\\";
export default (() => <>
        <p>{localize(\\"3\\")}</p>
        <Title id=\\"titleId\\" label=\\"lorem ipsum\\" />
    </>);
export const localizeKeyMap = {
  3: \\"hello world\\"
};"
`;

exports[`should set localizerSource 1`] = `
"import { localize } from \\"testlocalizerSource\\";
export default (() => <>
        <p>{localize(\\"2\\")}</p>
        <Title id=\\"titleId\\" label=\\"lorem ipsum\\" />
    </>);
export const localizeKeyMap = {
  2: \\"hello world\\"
};"
`;
